homeassistant/components/min_max/config_flow.py

```diff
@@ -72,3 +75,45 @@ class ConfigFlowHandler(SchemaConfigFlowHandler, domain=DOMAIN):
     def async_config_entry_title(self, options: Mapping[str, Any]) -> str:
         """Return config entry title."""
         return cast(str, options["name"]) if "name" in options else ""
+
+    @callback
+    @staticmethod
+    def async_setup_preview() -> None:
+        """Set up preview WS API."""
+        hass = async_get_hass()
+        websocket_api.async_register_command(hass, ws_preview_min_max)
+
+
+@callback
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "min_max/preview",
```

Comment:
Should we make this command more general? Why do we need the `flow_id` and `flow_type`, will there be a difference between options flow and config flow in the future? Why not just allow the `user_input`? Because of defaults that are not communicated in the schema?

Reply:
I think that in the end, these endpoint APIs for preview rely fully on what the frontend needs. If frontend can do it without the type it's fine.

I think here it's needed as not all data can be changed from an options flow, so you need to merge it with the full config.

Reply:
We need the `flow_id` and `flow_type` to be able to:
- access user input from previous steps
- access existing config entry settings in an options flow